kratos/geometries/coupling_geometry.h

```diff
@@ -192,20 +192,35 @@ template<class TPointType> class CouplingGeometry
     ///@{
 
     /**
-    * @brief This function returns the geometry part which is accessable
-    *        with a certain index.
-    * @details Here, 0 is the master, all above 0 is for the slaves
-    * of the geometry. Checks if index is available only in debug mode.
-    * @param Index of the geometry part. 0->Master; all bigger than 1->Slave
-    * @return geometry, which is connected through the Index
-     */
-    GeometryType& GetGeometryPart(IndexType Index) const override
+    * @brief This function returns the pointer of the geometry part
+    *        which is corresponding to the index.
+    *        Checks if index is available only in debug.
+    * @param Index: 0 -> master, all bigger than 0 -> slaves.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    GeometryPointer pGetGeometryPart(IndexType Index) override
+    {
+        KRATOS_DEBUG_ERROR_IF(mpGeometries.size() <= Index) << "Index "
+            << Index << " out of range. Composite contains only of: "
```

Comment:
Please revise the english

Reply:
I tried my best ;) hope it is better now.